In [4]:
#dependencies
import pandas as pd
import numpy as np

In [5]:
#file to load
file_to_load = "purchase_data.csv"
purchase_data= pd.read_csv(file_to_load,error_bad_lines=False)
#purchase_data2=purchase_data.copy()
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [6]:
purchase_data.shape

(780, 7)

In [7]:
#player count

total_player = len(purchase_data["SN"].unique())
total_player_df=pd.Series({
    "Total Players": total_player,
})
total_player_df

Total Players    576
dtype: int64

In [8]:
#Purchasing Analysis (total)

#Number of Unique Items
unique = len(purchase_data["Item ID"].unique())

#Average Purchase Price
average_purchase_price = purchase_data["Price"].mean()
average_purchase_price

#Total Number of Purchases
total_number_purchases = purchase_data["Purchase ID"].count()
total_number_purchases


#Total Revenue
total_revenue= purchase_data["Price"].sum()
total_revenue

purchasing_analysis_df= pd.DataFrame({
   "Number of Unique Items": unique,
    "Average Price": "$"+ str(average_purchase_price),
    "Number of Purchases": [total_number_purchases],
    "Total Revenue": "$"+ str(total_revenue)
})
purchasing_analysis_df

Number of Unique Items       Average Price  Number of Purchases  \
0                     183  $3.050987179487176                  780   

  Total Revenue  
0      $2379.77

In [9]:
#Gender Demographics
gender_demo=purchase_data.drop_duplicates(["SN"])["Gender"].value_counts()
avg_players=gender_demo*100/total_player
sum_gender=pd.DataFrame({
    "Total Count": gender_demo,
    "Percentage of Players": avg_players})
sum_gender

Total Count  Percentage of Players
Male                           484              84.027778
Female                          81              14.062500
Other / Non-Disclosed           11               1.909722

In [16]:
#Purchasing Analysis (Gender)
purchase_count=pd.DataFrame(purchase_data["Gender"].value_counts())

total_purchase_gender=purchase_data[["Gender", "Price"]]
total_purchase=total_purchase_gender.groupby(["Gender"]).sum()

avg_purchase_price=total_purchase["Price"]/purchase_count["Gender"]
#avg_purchase_price.columns=["Average Purchase Price"]
avg_ttl_purchase=total_purchase["Price"]/sum_gender["Total Count"]


sum_gender2=pd.DataFrame({
   "Purchase Count": purchase_count['Gender'],
   "Total purchase": total_purchase['Price'],
   "Average Purchase Price": avg_purchase_price,
   "Average TTL Price": avg_ttl_purchase
})
sum_gender2

Purchase Count  Total purchase  Average Purchase Price  \
Female                            113          361.94                3.203009   
Male                              652         1967.64                3.017853   
Other / Non-Disclosed              15           50.19                3.346000   

                       Average TTL Price  
Female                          4.468395  
Male                            4.065372  
Other / Non-Disclosed           4.562727

In [19]:
#Age Demographics
purchase_data.drop_duplicates(["SN"], inplace = True)
purchase_data["Categories"] = pd.DataFrame(pd.cut(purchase_data["Age"], [-np.inf,10,15,20,25,30,35,40,+np.inf],labels=['<10','10-14','15-19','20-24','25-29','30-34','35-39','40+'],right = False, include_lowest = True))
agecount = pd.DataFrame(purchase_data.groupby("Categories").count(), index =['<10','10-14','15-19','20-24','25-29','30-34','35-39','40+'])

summary_age = pd.DataFrame()
summary_age["Total Count"] = agecount["Purchase ID"]
summary_age["Percentage of Players"] = (summary_age["Total Count"] / summary_age["Total Count"].sum())*100
#summary_age2 = summary_age.copy()
#summaryAgeCpy.index.name = 'Tabulador'
summary_age

Total Count  Percentage of Players
<10             17               2.951389
10-14           22               3.819444
15-19          107              18.576389
20-24          258              44.791667
25-29           77              13.368056
30-34           52               9.027778
35-39           31               5.381944
40+             12               2.083333

In [26]:
purchase_data["Categories"] = pd.DataFrame(pd.cut(purchase_data["Age"], [-np.inf,10,15,20,25,30,35,40,+np.inf],labels=['<10','10-14','15-19','20-24','25-29','30-34','35-39','40+'],right = False, include_lowest = True))

groupages = pd.DataFrame (purchase_data.groupby('Categories').agg({'SN':'size', 
                         'Price':['mean','sum']}), index =['<10','10-14','15-19','20-24','25-29','30-34','35-39','40+'])
groupages.index.name = "abc"
groupages.columns= ['Purchase Count','Average Purchase Price','Total Purchase Value']

groupages = groupages.join(summary_age[['Total Count']], on="abc")
groupages["Avg Total Purchase per Person"]=groupages['Total Purchase Value']/groupages["Total Count"]
groupages = groupages.drop("Total Count",1)
groupages.index.name = ''
groupages

Purchase Count  Average Purchase Price  Total Purchase Value  \
                                                                      
<10                17                3.390000                 57.63   
10-14              22                3.074545                 67.64   
15-19             107                3.101682                331.88   
20-24             258                3.063527                790.39   
25-29              77                2.908182                223.93   
30-34              52                2.921538                151.92   
35-39              31                3.510000                108.81   
40+                12                3.037500                 36.45   

       Avg Total Purchase per Person  
                                      
<10                         3.390000  
10-14                       3.074545  
15-19                       3.101682  
20-24                       3.063527  
25-29                       2.908182  
30-34                       2.921538  
35-39                       3.510000  
40+                         3.037500

In [191]:
#Top Spenders
#top_spenders = purchase_data.sort_values("SN", ascending=False)
#top_spenders.head(5)


groupSpenders = purchase_data2.groupby('SN').agg({'SN':'size', 
                         'Price':['mean','sum']})
groupSpenders.columns= ['Purchase Count','Average Purchase Price','Total Purchase Value']
groupSpenders = groupSpenders.sort_values(['Total Purchase Value','Purchase Count'], ascending = False) 
groupSpenders.head()

Purchase Count  Average Purchase Price  Total Purchase Value
SN                                                                       
Lisosia93                 5                3.792000                 18.96
Idastidru52               4                3.862500                 15.45
Chamjask73                3                4.610000                 13.83
Iral74                    4                3.405000                 13.62
Iskadarya95               3                4.366667                 13.10

In [29]:
#Most Popular Items

mostpopu = purchase_data.groupby('Item ID').agg({'Item ID':'size',                                                       
                                                       'Price':'sum'})
mostpopu.columns= ['Purchase Count','Total Purchase Value']
mostpopu =mostpopu.sort_values(['Purchase Count','Item ID'], ascending = False) 

mostpopu = mostpopu.join(purchase_data[['Item Name', 'Price']], on='Item ID')

mostpopu.set_index(['Item Name'], inplace = True, append = True)

mostpopu = mostpopu[['Purchase Count','Price','Total Purchase Value']]

mostpopu .head()


Purchase Count  Price  \
Item ID Item Name                                                     
178     Despair, Favor of Due Diligence                   11   4.60   
37      Ritual Mace                                        8   2.05   
159     Betrayal, Whisper of Grieving Widows               7   3.94   
145     Hopeless Ebon Dualblade                            7   1.33   
82      Azurewrath                                         7   4.40   

                                              Total Purchase Value  
Item ID Item Name                                                   
178     Despair, Favor of Due Diligence                      46.53  
37      Ritual Mace                                          25.28  
159     Betrayal, Whisper of Grieving Widows                 21.56  
145     Hopeless Ebon Dualblade                              32.06  
82      Azurewrath                                           34.30

In [193]:
#Most Profitable Items
groupPopular = groupPopular.sort_values(['Total Purchase Value'], ascending = False) 
groupPopular.head()

Purchase Count  Price  \
Item ID Item Name                                                     
178     Despair, Favor of Due Diligence                   12   4.60   
82      Azurewrath                                         9   4.40   
145     Hopeless Ebon Dualblade                            9   1.33   
92      Betrayal, Whisper of Grieving Widows               8   3.94   
103     Thorn, Satchel of Dark Souls                       8   1.33   

                                              Total Purchase Value  
Item ID Item Name                                                   
178     Despair, Favor of Due Diligence                      50.76  
82      Azurewrath                                           44.10  
145     Hopeless Ebon Dualblade                              41.22  
92      Betrayal, Whisper of Grieving Widows                 39.04  
103     Thorn, Satchel of Dark Souls                         34.80

trend of the data:

Of the 1163 active players, the vast majority are male (84%). There also exists, a smaller, but notable proportion of female players (14%).
Our peak age demographic falls between 20-24 (44.8%) with secondary groups falling between 15-19 (18.60%) and 25-29 (13.4%).

In [ ]:
conlusion:
    